cmake/leveldb.cmake

```diff
@@ -67,27 +61,33 @@ target_compile_definitions(leveldb
     HAVE_O_CLOEXEC=$<BOOL:${HAVE_O_CLOEXEC}>
     FALLTHROUGH_INTENDED=[[fallthrough]]
     LEVELDB_IS_BIG_ENDIAN=${WORDS_BIGENDIAN}
+    $<$<NOT:$<BOOL:${WIN32}>>:LEVELDB_PLATFORM_POSIX>
+    $<$<BOOL:${WIN32}>:LEVELDB_PLATFORM_WINDOWS>
+    $<$<BOOL:${WIN32}>:_UNICODE;UNICODE>
+    $<$<BOOL:${MINGW}>:__USE_MINGW_ANSI_STDIO=1>
 )
 
-if(WIN32)
-  target_compile_definitions(leveldb
-    PRIVATE
-      LEVELDB_PLATFORM_WINDOWS
-      _UNICODE
-      UNICODE
-      __USE_MINGW_ANSI_STDIO=1
-  )
-else()
-  target_compile_definitions(leveldb PRIVATE LEVELDB_PLATFORM_POSIX)
-endif()
-
 target_include_directories(leveldb
   PRIVATE
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/leveldb>
   PUBLIC
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/leveldb/include>
 )
 
+if(MSVC)
+  target_compile_options(leveldb
+    PRIVATE
+      /wd4244
```

Comment:
Does it make sense to set these if they are set anyway in the top level `CMakeLists.txt`?

Reply:
At some point in the future, I might want to modify our codebase to fix warnings. It is unlikely that we want to touch subtree codebase for the same reason.